kafka/producer.go

```diff
@@ -572,6 +572,14 @@ func (p *Producer) QueryWatermarkOffsets(topic string, partition int32, timeoutM
 	return queryWatermarkOffsets(p, topic, partition, timeoutMs)
 }
 
+// GetConsumerGroupMetadata queries broker for consumer groups metadata and all its members
+// like which consumer groups and which corresponding members consume which topics
+// if consumerGroup is nil, metadata for all consumer groups will be returned
+// otherwise only metadata for that specific consumer group will be returned
+func (p *Producer) GetConsumerGroupMetadata(consumerGroup *string, timeoutMs int) ([]ConsumerGroupMetadata, error) {
```

Comment:
We don't need to provide this method on Producer and Consumer since an AdminClient can be instantiated from an existing Producer/Consumer without much cost.